packages/realm/src/binding/NativeBigInt.ts

```diff
@@ -0,0 +1,40 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2024 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import type { binding } from "./wrapper.generated";
+
+export const NativeBigInt: typeof binding.Int64 = Object.freeze({
+  add(a: binding.Int64, b: binding.Int64) {
+    return a + b;
+  },
+  equals(a: binding.Int64, b: binding.Int64) {
```

Comment:
To conform to prior behavior:
```suggestion
  equals(a: binding.Int64, b: binding.Int64 | number | string) {
```

Reply:
I do see it mentioned here: https://github.com/realm/realm-js/blob/ff3d889df4bb1da1ae40074cbcbea8bf8410223f/packages/realm/bindgen/src/templates/typescript.ts#L201C32-L201C50 but I wonder if anything is relying on it or if it would be okay to simplify 🤔 I mean, what is the expected semantics here? For one thing, must the two things being compared match in type or will they get coerced?

Reply:
I think the ambiguity here is what type of equality our `equals()` method is supposed to be. The previous behavior mimics loose equality, so if it's a number, their numeric values are compared; whereas if it's a string, the string gets converted to a `BigInt`.